.github/workflows/windows.yml

```diff
@@ -413,52 +413,125 @@ jobs:
+          restore-keys: ${{ matrix.plat }}-${{ matrix.name }}
+
+      - name: 'vcpkg init'
+        if: success()
+        shell: bash
+        run: |
+          git clone https://github.com/microsoft/vcpkg/ --depth 1
+          vcpkg/bootstrap-vcpkg.sh
+
+      - name: 'vcpkg build'
+        shell: bash
+        run: |
+          export VCPKG_ROOT="$PWD/vcpkg"
+          export CURRENT_BINARY_CACHE='cache/${{ matrix.name }}'
+          mkdir -p "$CURRENT_BINARY_CACHE"
+          export VCPKG_BINARY_SOURCES="clear;files,$GITHUB_WORKSPACE/$CURRENT_BINARY_CACHE,readwrite;"
+          vcpkg/vcpkg x-set-installed ${{ matrix.install }} --triplet=${{ matrix.arch }}-${{ matrix.plat }}
+
+      - name: 'save cache dependencies'
+        id: cache-save
+        uses: actions/cache/save@v4
+        with:
+          path: 'cache/${{ matrix.name }}'
+          key: ${{ matrix.plat }}-${{ matrix.name }}-${{ hashFiles('vcpkg/installed/vcpkg/info/*', 'vcpkg/installed/vcpkg/updates/*') }}
+
       - name: 'cmake configure'
         timeout-minutes: 5
         shell: bash
         run: |
-          archgen=${{ matrix.arch }}; [ "${archgen}" = 'x86' ] && archgen='Win32'
           if [ '${{ matrix.plat }}' = 'uwp' ]; then
             system='WindowsStore'
             options='-DCMAKE_SYSTEM_VERSION=10.0'
           else
             system='Windows'
           fi
-          [ '${{ matrix.type }}' = 'Debug' ] && options+=' -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG='
-          [ '${{ matrix.type }}' = 'Release' ] && options+=' -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE='
-          cmake -B bld ${options} \
-            "-DCMAKE_SYSTEM_NAME=${system}" \
-            -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake \
-            "-DCMAKE_GENERATOR_PLATFORM=${archgen}" \
-            '-DVCPKG_TARGET_TRIPLET=${{ matrix.arch }}-${{ matrix.plat }}' \
+          export VCPKG_ROOT="$PWD/vcpkg"
+          cmake . -B bld ${options} \
+            -DCMAKE_TOOLCHAIN_FILE=$VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake \
+            -DVCPKG_INSTALLED_DIR=$VCPKG_ROOT/installed \
+            -DVCPKG_TARGET_TRIPLET=${{ matrix.arch }}-${{ matrix.plat }} \
+            -DCMAKE_BUILD_TYPE=${{ matrix.type }} \
+            -DCMAKE_SYSTEM_NAME=$system \
             -DCMAKE_VS_GLOBALS=TrackFileAccess=false \
-            '-DCMAKE_UNITY_BUILD=${{ matrix.unity }}' \
-            "-DCMAKE_C_FLAGS=${cflags}" \
-            '-DCMAKE_BUILD_TYPE=${{ matrix.type }}' \
             -DCMAKE_UNITY_BUILD=ON \
             -DCURL_WERROR=ON \
             -DBUILD_EXAMPLES=ON \
             -DENABLE_WEBSOCKETS=ON \
+            -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG= \
+            -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE= \
             ${{ matrix.config }} || { cat bld/CMakeFiles/CMake*.yaml; false; }
 
+      - name: 'prepare logs on failure'
+        if: failure()
+        shell: bash
+        run: |
+          export VCPKG_ROOT="$PWD/vcpkg"
+          mkdir -p bld
```

Comment:
Not always. It can fail only the vcpkg build without the cmake generation.

Reply:
Yeah, but then all the bld files referenced from 7z would be missing also. Is there any reason to have an empty `bld` directory there?

Local tests say that 7z fails with 127 anyway. This makes it seems that a ` || true` is what's needed and the `mkdir` is redundant.

Reply:
We have 2 cases.
First case: `vcpkg build`. In this case there is $VCPKG_ROOT/buildtrees/*.log.
If you will put  || true it will fail on create the log file, and upload nothing. 
On my case it will create empty dir, will upload   $VCPKG_ROOT/buildtrees/*.log and all other files, it will give warning that it not find them, it create the log.7z and upload it.

In the second case: cmake configure
The folder already exsit. the mkdir will not create folder and not fail. (because of the -p argument).
It will find the files and upload them.

Reply:
Ok, got it and could also reproduce locally.

Is the `-p` option needed though? We're only creating a single-level subdir here.

Reply:
Yes, because the folder can be exist. if you make a dir with exist folder without `-p` argument it fail. you can test it local on the bash:
```bash
mkdir tal
mkdir tal
mkdir: cannot create directory ‘tal’: File exists
```